lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -475,26 +473,30 @@ def detach_eni(eni, module):
         module.exit_json(changed=False, interface=get_eni_info(eni))
 
 
-def find_eni(connection, module):
+def uniquely_find_eni(connection, module):
 
     eni_id = module.params.get("eni_id")
-    subnet_id = module.params.get('subnet_id')
     private_ip_address = module.params.get('private_ip_address')
     instance_id = module.params.get('instance_id')
     device_index = module.params.get('device_index')
 
     try:
         filters = {}
-        if subnet_id:
```

Comment:
Oh, but I guess I'm not sure about this for another reason. private_ip_address isn't unique. Isn't more filtering better in a case like that?